src/pages/Validators/ValidatorsMap.tsx

```diff
@@ -1,38 +1,71 @@
-import React from "react";
+import React, {useEffect, useState} from "react";
 import {Stack, useMediaQuery, useTheme} from "@mui/material";
 import Map from "./Components/Map";
 import {grey} from "../../themes/colors/aptosColorPalette";
 import MapMetrics from "./Components/MapMetrics";
 import {useGetValidatorSetGeoData} from "../../api/hooks/useGetValidatorSetGeoData";
+import {useGetEpochTime} from "../../api/hooks/useGetEpochTime";
+import {useGetValidatorSet} from "../../api/hooks/useGetValidatorSet";
+import {useGetStakingRewardsRate} from "../../api/hooks/useGetStakingRewardsRate";
+import {SkeletonTheme} from "react-loading-skeleton";
 
 export default function ValidatorsMap() {
   const theme = useTheme();
+  const isDarkTheme = theme.palette.mode === "dark";
   const isOnMobile = !useMediaQuery(theme.breakpoints.up("md"));
-  const backgroundColor = theme.palette.mode === "dark" ? grey[800] : grey[50];
+  const backgroundColor = isDarkTheme ? grey[800] : grey[50];
 
   const {validatorGeoMetric, validatorGeoGroups} = useGetValidatorSetGeoData();
+  const [isSkeletonLoading, setIsSkeletonLoading] = useState<boolean>(true);
+  const {curEpoch} = useGetEpochTime();
+  const {totalVotingPower} = useGetValidatorSet();
+  const {rewardsRateYearly} = useGetStakingRewardsRate();
+  const {numberOfActiveValidators} = useGetValidatorSet();
 
-  return isOnMobile ? (
-    <Stack
-      direction="column"
-      justifyContent="space-between"
-      marginY={4}
-      sx={{backgroundColor: backgroundColor}}
-      overflow="hidden"
-    >
-      <Map validatorGeoGroups={validatorGeoGroups} />
-      <MapMetrics validatorGeoMetric={validatorGeoMetric} isOnMobile />
-    </Stack>
-  ) : (
-    <Stack
-      direction="row"
-      justifyContent="space-between"
-      marginY={4}
-      sx={{backgroundColor: backgroundColor}}
-      overflow="hidden"
-    >
-      <MapMetrics validatorGeoMetric={validatorGeoMetric} />
-      <Map validatorGeoGroups={validatorGeoGroups} />
-    </Stack>
+  useEffect(() => {
+    if (
+      curEpoch &&
+      totalVotingPower &&
+      rewardsRateYearly &&
+      numberOfActiveValidators
+    ) {
+      setIsSkeletonLoading(false);
+    }
+  }, [curEpoch, totalVotingPower, rewardsRateYearly, numberOfActiveValidators]);
+
+  return (
+    <SkeletonTheme baseColor={isDarkTheme ? grey[500] : undefined}>
+      {isOnMobile ? (
+        <Stack
+          direction="column"
+          justifyContent="space-between"
+          marginY={4}
+          sx={{backgroundColor: backgroundColor}}
+          overflow="hidden"
+        >
+          <Map validatorGeoGroups={validatorGeoGroups} />
+          <MapMetrics
+            validatorGeoMetric={validatorGeoMetric}
+            isOnMobile={isOnMobile}
```

Comment:
```suggestion
            isOnMobile
```

Reply:
the reason why i specify the prop value here even though they have same naming, is cuz if i don't do so, the child component doesn't seem to be picking up parent's prop value change, see this attachment : ![Screenshot 2023-01-05 at 4 45 49 PM](https://user-images.githubusercontent.com/121921928/210907173-259c829e-1f9f-41a2-8345-754198a1e32d.png). the prop we pass down to MapMetrics was not updated accordingly.

Reply:
Interesting. This shouldn't happen. It worked well before. see https://github.com/aptos-labs/explorer/blob/main/src/pages/Validators/ValidatorsMap.tsx#L24. Can you dig deeper and try to see why? It's not a blocker though.

Reply:
FYI, `<Component isOnMobile>` is shorthand for `<Component isOnMobile={true}>`, not `<Component isOnMobile={isOnMobile}>`. Think HTML boolean attributes, not ES property names.